### PR TITLE
Added Polaris_AuthenticateTo() to allow use of an alternate API server.

### PR DESCRIPTION
**Pull request by adamshapiro0**

### CODE REVIEW

File: c/src/point_one/polaris/polaris.c [hunk 1]
```diff
@@ -210,16 +216,16 @@ int Polaris_Authenticate(PolarisContext_t* context, const char* api_key,
     return POLARIS_NOT_ENOUGH_SPACE;
   }
 
-  P1_DebugPrint("Sending auth request. [api_key=%s, unique_id=%s]\n", api_key,
-             unique_id);
+  P1_DebugPrint("Sending auth request. [api_key=%s, unique_id=%s, url=%s]\n",
+                api_key, api_url, unique_id);
```
Comment (anathan) [inline c/src/point_one/polaris/polaris.c:220]: Order of elements is swapped

File: c/src/point_one/polaris/polaris.c [hunk 2]
```diff
@@ -404,6 +410,7 @@ void Polaris_SetRTCMCallback(PolarisContext_t* context,
 /******************************************************************************/
 int Polaris_SendECEFPosition(PolarisContext_t* context, double x_m, double y_m,
                              double z_m) {
+  // TOOD Buidl in some sort of min time between position updates so they don't flood
```
Comment (anathan) [inline c/src/point_one/polaris/polaris.c:413]: Spelling

**Review by anathan — APPROVED** [review]
This works. I would have perhaps considered making the API URL part of the context instead to avoid the "three strings" argument in authenticate to, but this functionally works.

**Comment by adamshapiro0** [comment]
I thought about that a bit. To do it, we would still need to create a setter to modify the context since ideally we don't want anyone to mess with it directly in case it changes in the future. Even then, it would either store `const char*`, which would only work if the caller guaranteed the string didn't go out of scope ever, or have to store a char array, which increases memory footprint. With `const char*`, if they user used a temp string (e.g., wanted to use `sprintf()` or something) and it went out of scope, future calls to `Connect()` to handle reconnects would crash.

I figured this was the clearest and safest solution, and it's consistent with the existing `ConnectTo()` call.

File: c/src/point_one/polaris/polaris.c (hunk 2 above)
Comment (adamshapiro0) [reply c/src/point_one/polaris/polaris.c:413]: Oops didn't meant to commit this.

**Pull request merged**